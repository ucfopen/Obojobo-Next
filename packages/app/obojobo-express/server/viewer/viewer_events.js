const oboEvents = oboRequire('server/obo_events')
const viewerState = oboRequire('server/viewer/viewer_state')
const VisitModel = oboRequire('server/models/visit')
const db = oboRequire('/server/db')
const logger = oboRequire('/server/logger')

oboEvents.on('client:nav:open', event => {
	return setNavOpen(event.userId, event.draftId, event.contentId, true, event.visitId)
})

oboEvents.on('client:nav:close', event => {
	return setNavOpen(event.userId, event.draftId, event.contentId, false, event.visitId)
})

oboEvents.on('client:nav:setRedAlert', event => {
	const userId = event.userId
	const draftId = event.draftId
	const isRedAlertEnabled = event.payload.to
	return db.none(
		`
				INSERT INTO red_alert_status
				(user_id, draft_id, is_enabled)
				VALUES($[userId], $[draftId], $[isRedAlertEnabled])
				ON CONFLICT (user_id, draft_id) DO UPDATE
				SET is_enabled = $[isRedAlertEnabled]
				WHERE red_alert_status.user_id = $[userId] AND
				red_alert_status.draft_id = $[draftId]
			`,
		{
			userId,
			draftId,
			isRedAlertEnabled
		}
	)
})

const setNavOpen = (userId, draftId, contentId, value, visitId) => {
	return VisitModel.fetchById(visitId).then(visit => {
		viewerState.set(userId, draftId, contentId, 'nav:isOpen', 1, value, visit.resource_link_id)
	})
}

module.exports = (req, res, next) => {
	next()
}
