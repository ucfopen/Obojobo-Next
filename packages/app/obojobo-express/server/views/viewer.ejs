<!DOCTYPE html>
<html lang="en">

<head>
	<%
			const title = 'Obojobo Next Document Viewer'
			const css = [
				webpackAssetPath('viewer.css'),
			]
			const headerJs = []
			const footerJs = [
				assetForEnv('//unpkg.com/react@16.13.1/umd/react.$[development|production.min].js'),
				assetForEnv('//unpkg.com/react-dom@16.13.1/umd/react-dom.$[development|production.min].js'),
				assetForEnv('//unpkg.com/underscore@1.11.0/underscore$[-min].js'),
				assetForEnv('https://unpkg.com/backbone@1.3.3/backbone$[-min].js'),
				webpackAssetPath('viewer.js')
			]
			const fonts = ['//fonts.googleapis.com/css?family=Libre+Franklin:400,400i,700,700i,900,900i|Roboto+Mono:400,400i,700,700i|Noto+Serif:400,400i,700,700i']
		%>
		<%- include('./partials/head', {title, css, fonts, headerJs}) %>
</head>

<body>
	<div id="viewer-app"></div>
	<div id="viewer-app-loading">
		<%- include('./partials/loading') %>
	</div>
	<%
		if (typeof inlines !== 'undefined')
		{
			inlines.forEach(function(inline)
			{ %>
		<script><% - js %></script>
		<% })
		}

		if (typeof footerJs !== 'undefined')
		{
			footerJs.forEach(function(href) { %>
			<script src="<%= href %>"></script>
			<% })
		}
	%>
				<script>window.__oboViewerRender()</script>
</body>

</html>
