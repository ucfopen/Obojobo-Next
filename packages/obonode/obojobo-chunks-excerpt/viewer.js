import Common from 'obojobo-document-engine/src/scripts/common'
import ViewerComponent from './viewer-component'
import adapter from './adapter'

Common.Registry.registerModel('ObojoboDraft.Chunks.Excerpt', {
	adapter,
	componentClass: ViewerComponent,
	default: true,
	type: 'chunk'
})
