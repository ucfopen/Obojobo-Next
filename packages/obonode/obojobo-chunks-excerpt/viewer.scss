@import '../obojobo-document-engine/src/scss';

.obojobo-draft--chunks--excerpt {
	pre {
		font-size: 1em;
		padding-left: 2em;
		padding-right: 2em;
	}

	.text {
		display: block;
	}
}
