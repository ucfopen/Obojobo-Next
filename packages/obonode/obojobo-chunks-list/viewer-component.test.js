import React from 'react'
import renderer from 'react-test-renderer'

import List from './viewer-component'
import OboModel from 'obojobo-document-engine/src/scripts/common/models/obo-model'

require('./viewer') // used to register this oboModel

describe('List', () => {
	test('List component', () => {
		const moduleData = {
			focusState: {}
		}
		const model = OboModel.create({
			id: 'id',
			type: 'ObojoboDraft.Chunks.List',
			content: {
				textGroup: [],
				listStyles: {
					type: 'ordered',
					indents: {
						'2': {
							type: 'unordered',
							bulletStyle: 'square'
						},
						'4': {
							type: 'ordered',
							start: '10',
							bulletStyle: 'upper-alpha'
						}
					}
				}
			}
		})

		const component = renderer.create(<List model={model} moduleData={moduleData} />)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('List component with ascending indents', () => {
		const moduleData = {
			focusState: {}
		}
		const model = OboModel.create({
			id: 'id',
			type: 'ObojoboDraft.Chunks.List',
			content: {
				textGroup: [
					{
						text: {
							value: 'One(indent=1)',
							styleList: []
						},
						data: {
							indent: '1'
						}
					},
					{
						text: {
							value: 'Two(indent=2)',
							styleList: []
						},
						data: {
							indent: '2'
						}
					},
					{
						text: {
							value: 'Three(indent=3)',
							styleList: []
						},
						data: {
							indent: '3'
						}
					},
					{
						text: {
							value: 'Four(indent=4)',
							styleList: []
						},
						data: {
							indent: '4'
						}
					},
					{
						text: {
							value: 'Five(indent=5)',
							styleList: []
						},
						data: {
							indent: '5'
						}
					},
					{
						text: {
							value: 'Six(indent=6)',
							styleList: []
						},
						data: {
							indent: '6'
						}
					}
				],
				listStyles: {
					type: 'ordered',
					indents: {
						'2': {
							type: 'unordered',
							bulletStyle: 'square'
						},
						'4': {
							type: 'ordered',
							start: '10',
							bulletStyle: 'upper-alpha'
						}
					}
				}
			}
		})

		const component = renderer.create(<List model={model} moduleData={moduleData} />)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('List component with inverse indentation', () => {
		const moduleData = {
			focusState: {}
		}
		const model = OboModel.create({
			id: 'id',
			type: 'ObojoboDraft.Chunks.List',
			content: {
				textGroup: [
					{
						text: {
							value: 'One(indent=1)',
							styleList: []
						},
						data: '3'
					},
					{
						text: {
							value: 'Two(indent=2)',
							styleList: []
						},
						data: {
							indent: '2'
						}
					},
					{
						text: {
							value: 'Three(indent=6)',
							styleList: []
						},
						data: {
							indent: '1'
						}
					}
				],
				listStyles: {
					type: 'ordered',
					indents: {
						'2': {
							type: 'unordered',
							bulletStyle: 'square'
						},
						'4': {
							type: 'ordered',
							start: '10',
							bulletStyle: 'upper-alpha'
						}
					}
				}
			}
		})

		const component = renderer.create(<List model={model} moduleData={moduleData} />)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('List component with irregular indentation', () => {
		const moduleData = {
			focusState: {}
		}
		const model = OboModel.create({
			id: 'id',
			type: 'ObojoboDraft.Chunks.List',
			content: {
				textGroup: [
					{
						text: {
							value: 'One(indent=1)',
							styleList: []
						},
						data: null
					},
					{
						text: {
							value: 'Two(indent=2)',
							styleList: []
						},
						data: {
							indent: '2'
						}
					},
					{
						text: {
							value: 'Three(indent=6)',
							styleList: []
						},
						data: {
							indent: '6'
						}
					},
					{
						text: {
							value: 'Four(indent=2)',
							styleList: []
						},
						data: {
							indent: '2'
						}
					},
					{
						text: {
							value: 'Five(indent=4)',
							styleList: []
						},
						data: {
							indent: '4'
						}
					},
					{
						text: {
							value: 'Six(indent=3)',
							styleList: []
						},
						data: {
							indent: '3'
						}
					}
				],
				listStyles: {
					type: 'ordered',
					indents: {
						'2': {
							type: 'unordered',
							bulletStyle: 'square'
						},
						'4': {
							type: 'ordered',
							start: '10',
							bulletStyle: 'upper-alpha'
						}
					}
				}
			}
		})

		const component = renderer.create(<List model={model} moduleData={moduleData} />)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('List component with colored Text', () => {
		const moduleData = {
			focusState: {}
		}
		const model = OboModel.create({
			id: 'id',
			type: 'ObojoboDraft.Chunks.List',
			content: {
				textGroup: [
					{
						data: {
							indent: 0
						},
						text: {
							value: 'Title',
							styleList: [
								{
									end: 5,
									data: {
										text: '#1155cc'
									},
									type: 'color',
									start: 0
								}
							]
						}
					},
					{
						data: {
							indent: 1
						},
						text: {
							value: 'Indent',
							styleList: [
								{
									end: 6,
									data: {
										text: '#cc0000'
									},
									type: 'color',
									start: 0
								}
							]
						}
					},
					{
						data: {
							indent: 0
						},
						text: {
							value: 'Another Title',
							styleList: [
								{
									end: 13,
									data: {
										text: '#f1c232'
									},
									type: 'color',
									start: 0
								}
							]
						}
					},
					{
						data: {
							indent: 1
						},
						text: {
							value: 'Another Indent',
							styleList: [
								{
									end: 14,
									data: {
										text: '#6aa84f'
									},
									type: 'color',
									start: 0
								}
							]
						}
					}
				],
				listStyles: {
					type: 'unordered',
					indents: {
						'0': {
							type: 'unordered',
							bulletStyle: 'disc'
						},
						'1': {
							type: 'unordered',
							bulletStyle: 'circle'
						}
					}
				}
			}
		})

		const component = renderer.create(<List model={model} moduleData={moduleData} />)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})
})
