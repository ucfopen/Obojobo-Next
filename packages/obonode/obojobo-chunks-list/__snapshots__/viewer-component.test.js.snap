// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List List component 1`] = `
<div
  className="component"
  data-focussed={false}
  data-id="id"
  data-obo-component={true}
  data-type="ObojoboDraft.Chunks.List"
  id="obo-id"
  tabIndex={-1}
>
  <div
    className="text-chunk obojobo-draft--chunks--list pad"
  >
    <div>
      <ol
        start={1}
        style={
          Object {
            "color": "",
            "listStyleType": undefined,
          }
        }
      >
        <li
          style={
            Object {
              "color": "",
              "listStyleType": "none",
            }
          }
        />
      </ol>
    </div>
  </div>
</div>
`;

exports[`List List component with ascending indents 1`] = `
<div
  className="component"
  data-focussed={false}
  data-id="id"
  data-obo-component={true}
  data-type="ObojoboDraft.Chunks.List"
  id="obo-id"
  tabIndex={-1}
>
  <div
    className="text-chunk obojobo-draft--chunks--list pad"
  >
    <div>
      <ol
        start={1}
        style={
          Object {
            "color": "",
            "listStyleType": undefined,
          }
        }
      >
        <li
          style={
            Object {
              "color": "",
              "listStyleType": "none",
            }
          }
        >
          <ol
            start={1}
            style={
              Object {
                "color": "",
                "listStyleType": undefined,
              }
            }
          >
            <li
              style={
                Object {
                  "color": "",
                  "listStyleType": "lower-alpha",
                }
              }
            >
              <span
                className="obo-text"
                data-group-index={0}
              >
                <span>
                  <span>
                    <span>
                      One(indent=1)
                    </span>
                  </span>
                </span>
              </span>
              <ul
                start={1}
                style={
                  Object {
                    "color": "",
                    "listStyleType": undefined,
                  }
                }
              >
                <li
                  style={
                    Object {
                      "color": "",
                      "listStyleType": "square",
                    }
                  }
                >
                  <span
                    className="obo-text"
                    data-group-index={1}
                  >
                    <span>
                      <span>
                        <span>
                          Two(indent=2)
                        </span>
                      </span>
                    </span>
                  </span>
                  <ol
                    start={1}
                    style={
                      Object {
                        "color": "",
                        "listStyleType": undefined,
                      }
                    }
                  >
                    <li
                      style={
                        Object {
                          "color": "",
                          "listStyleType": "upper-alpha",
                        }
                      }
                    >
                      <span
                        className="obo-text"
                        data-group-index={2}
                      >
                        <span>
                          <span>
                            <span>
                              Three(indent=3)
                            </span>
                          </span>
                        </span>
                      </span>
                      <ol
                        start="10"
                        style={
                          Object {
                            "color": "",
                            "listStyleType": undefined,
                          }
                        }
                      >
                        <li
                          style={
                            Object {
                              "color": "",
                              "listStyleType": "upper-alpha",
                            }
                          }
                        >
                          <span
                            className="obo-text"
                            data-group-index={3}
                          >
                            <span>
                              <span>
                                <span>
                                  Four(indent=4)
                                </span>
                              </span>
                            </span>
                          </span>
                          <ol
                            start={1}
                            style={
                              Object {
                                "color": "",
                                "listStyleType": undefined,
                              }
                            }
                          >
                            <li
                              style={
                                Object {
                                  "color": "",
                                  "listStyleType": "decimal",
                                }
                              }
                            >
                              <span
                                className="obo-text"
                                data-group-index={4}
                              >
                                <span>
                                  <span>
                                    <span>
                                      Five(indent=5)
                                    </span>
                                  </span>
                                </span>
                              </span>
                              <ol
                                start={1}
                                style={
                                  Object {
                                    "color": "",
                                    "listStyleType": undefined,
                                  }
                                }
                              >
                                <li
                                  style={
                                    Object {
                                      "color": "",
                                      "listStyleType": "lower-alpha",
                                    }
                                  }
                                >
                                  <span
                                    className="obo-text"
                                    data-group-index={5}
                                  >
                                    <span>
                                      <span>
                                        <span>
                                          Six(indent=6)
                                        </span>
                                      </span>
                                    </span>
                                  </span>
                                </li>
                              </ol>
                            </li>
                          </ol>
                        </li>
                      </ol>
                    </li>
                  </ol>
                </li>
              </ul>
            </li>
          </ol>
        </li>
      </ol>
    </div>
  </div>
</div>
`;

exports[`List List component with colored Text 1`] = `
<div
  className="component"
  data-focussed={false}
  data-id="id"
  data-obo-component={true}
  data-type="ObojoboDraft.Chunks.List"
  id="obo-id"
  tabIndex={-1}
>
  <div
    className="text-chunk obojobo-draft--chunks--list pad"
  >
    <div>
      <ul
        start={1}
        style={
          Object {
            "color": "",
            "listStyleType": undefined,
          }
        }
      >
        <li
          style={
            Object {
              "color": "#1155cc",
              "listStyleType": "disc",
            }
          }
        >
          <span
            className="obo-text"
            data-group-index={0}
          >
            <span>
              <span>
                <span
                  style={
                    Object {
                      "color": "#1155cc",
                    }
                  }
                >
                  <span>
                    Title
                  </span>
                </span>
              </span>
            </span>
          </span>
          <ul
            start={1}
            style={
              Object {
                "color": "",
                "listStyleType": undefined,
              }
            }
          >
            <li
              style={
                Object {
                  "color": "#cc0000",
                  "listStyleType": "circle",
                }
              }
            >
              <span
                className="obo-text"
                data-group-index={1}
              >
                <span>
                  <span>
                    <span
                      style={
                        Object {
                          "color": "#cc0000",
                        }
                      }
                    >
                      <span>
                        Indent
                      </span>
                    </span>
                  </span>
                </span>
              </span>
            </li>
          </ul>
        </li>
        <li
          style={
            Object {
              "color": "#f1c232",
              "listStyleType": "disc",
            }
          }
        >
          <span
            className="obo-text"
            data-group-index={2}
          >
            <span>
              <span>
                <span
                  style={
                    Object {
                      "color": "#f1c232",
                    }
                  }
                >
                  <span>
                    Another Title
                  </span>
                </span>
              </span>
            </span>
          </span>
          <ul
            start={1}
            style={
              Object {
                "color": "",
                "listStyleType": undefined,
              }
            }
          >
            <li
              style={
                Object {
                  "color": "#6aa84f",
                  "listStyleType": "circle",
                }
              }
            >
              <span
                className="obo-text"
                data-group-index={3}
              >
                <span>
                  <span>
                    <span
                      style={
                        Object {
                          "color": "#6aa84f",
                        }
                      }
                    >
                      <span>
                        Another Indent
                      </span>
                    </span>
                  </span>
                </span>
              </span>
            </li>
          </ul>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`List List component with inverse indentation 1`] = `
<div
  className="component"
  data-focussed={false}
  data-id="id"
  data-obo-component={true}
  data-type="ObojoboDraft.Chunks.List"
  id="obo-id"
  tabIndex={-1}
>
  <div
    className="text-chunk obojobo-draft--chunks--list pad"
  >
    <div>
      <ol
        start={1}
        style={
          Object {
            "color": "",
            "listStyleType": undefined,
          }
        }
      >
        <li
          style={
            Object {
              "color": "",
              "listStyleType": "decimal",
            }
          }
        >
          <span
            className="obo-text"
            data-group-index={0}
          >
            <span>
              <span>
                <span>
                  One(indent=1)
                </span>
              </span>
            </span>
          </span>
          <ol
            start={1}
            style={
              Object {
                "color": "",
                "listStyleType": undefined,
              }
            }
          >
            <li
              style={
                Object {
                  "color": "",
                  "listStyleType": "none",
                }
              }
            >
              <ul
                start={1}
                style={
                  Object {
                    "color": "",
                    "listStyleType": undefined,
                  }
                }
              >
                <li
                  style={
                    Object {
                      "color": "",
                      "listStyleType": "square",
                    }
                  }
                >
                  <span
                    className="obo-text"
                    data-group-index={1}
                  >
                    <span>
                      <span>
                        <span>
                          Two(indent=2)
                        </span>
                      </span>
                    </span>
                  </span>
                </li>
              </ul>
            </li>
            <li
              style={
                Object {
                  "color": "",
                  "listStyleType": "lower-alpha",
                }
              }
            >
              <span
                className="obo-text"
                data-group-index={2}
              >
                <span>
                  <span>
                    <span>
                      Three(indent=6)
                    </span>
                  </span>
                </span>
              </span>
            </li>
          </ol>
        </li>
      </ol>
    </div>
  </div>
</div>
`;

exports[`List List component with irregular indentation 1`] = `
<div
  className="component"
  data-focussed={false}
  data-id="id"
  data-obo-component={true}
  data-type="ObojoboDraft.Chunks.List"
  id="obo-id"
  tabIndex={-1}
>
  <div
    className="text-chunk obojobo-draft--chunks--list pad"
  >
    <div>
      <ol
        start={1}
        style={
          Object {
            "color": "",
            "listStyleType": undefined,
          }
        }
      >
        <li
          style={
            Object {
              "color": "",
              "listStyleType": "decimal",
            }
          }
        >
          <span
            className="obo-text"
            data-group-index={0}
          >
            <span>
              <span>
                <span>
                  One(indent=1)
                </span>
              </span>
            </span>
          </span>
          <ol
            start={1}
            style={
              Object {
                "color": "",
                "listStyleType": undefined,
              }
            }
          >
            <li
              style={
                Object {
                  "color": "",
                  "listStyleType": "none",
                }
              }
            >
              <ul
                start={1}
                style={
                  Object {
                    "color": "",
                    "listStyleType": undefined,
                  }
                }
              >
                <li
                  style={
                    Object {
                      "color": "",
                      "listStyleType": "square",
                    }
                  }
                >
                  <span
                    className="obo-text"
                    data-group-index={1}
                  >
                    <span>
                      <span>
                        <span>
                          Two(indent=2)
                        </span>
                      </span>
                    </span>
                  </span>
                  <ol
                    start={1}
                    style={
                      Object {
                        "color": "",
                        "listStyleType": undefined,
                      }
                    }
                  >
                    <li
                      style={
                        Object {
                          "color": "",
                          "listStyleType": "none",
                        }
                      }
                    >
                      <ol
                        start="10"
                        style={
                          Object {
                            "color": "",
                            "listStyleType": undefined,
                          }
                        }
                      >
                        <li
                          style={
                            Object {
                              "color": "",
                              "listStyleType": "none",
                            }
                          }
                        >
                          <ol
                            start={1}
                            style={
                              Object {
                                "color": "",
                                "listStyleType": undefined,
                              }
                            }
                          >
                            <li
                              style={
                                Object {
                                  "color": "",
                                  "listStyleType": "none",
                                }
                              }
                            >
                              <ol
                                start={1}
                                style={
                                  Object {
                                    "color": "",
                                    "listStyleType": undefined,
                                  }
                                }
                              >
                                <li
                                  style={
                                    Object {
                                      "color": "",
                                      "listStyleType": "lower-alpha",
                                    }
                                  }
                                >
                                  <span
                                    className="obo-text"
                                    data-group-index={2}
                                  >
                                    <span>
                                      <span>
                                        <span>
                                          Three(indent=6)
                                        </span>
                                      </span>
                                    </span>
                                  </span>
                                </li>
                              </ol>
                            </li>
                          </ol>
                        </li>
                      </ol>
                    </li>
                  </ol>
                </li>
                <li
                  style={
                    Object {
                      "color": "",
                      "listStyleType": "square",
                    }
                  }
                >
                  <span
                    className="obo-text"
                    data-group-index={3}
                  >
                    <span>
                      <span>
                        <span>
                          Four(indent=2)
                        </span>
                      </span>
                    </span>
                  </span>
                  <ol
                    start={1}
                    style={
                      Object {
                        "color": "",
                        "listStyleType": undefined,
                      }
                    }
                  >
                    <li
                      style={
                        Object {
                          "color": "",
                          "listStyleType": "none",
                        }
                      }
                    >
                      <ol
                        start="10"
                        style={
                          Object {
                            "color": "",
                            "listStyleType": undefined,
                          }
                        }
                      >
                        <li
                          style={
                            Object {
                              "color": "",
                              "listStyleType": "upper-alpha",
                            }
                          }
                        >
                          <span
                            className="obo-text"
                            data-group-index={4}
                          >
                            <span>
                              <span>
                                <span>
                                  Five(indent=4)
                                </span>
                              </span>
                            </span>
                          </span>
                        </li>
                      </ol>
                    </li>
                    <li
                      style={
                        Object {
                          "color": "",
                          "listStyleType": "upper-alpha",
                        }
                      }
                    >
                      <span
                        className="obo-text"
                        data-group-index={5}
                      >
                        <span>
                          <span>
                            <span>
                              Six(indent=3)
                            </span>
                          </span>
                        </span>
                      </span>
                    </li>
                  </ol>
                </li>
              </ul>
            </li>
          </ol>
        </li>
      </ol>
    </div>
  </div>
</div>
`;
